charts/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -236,17 +236,17 @@ spec:
           {{- end }}
           - name: DD_REMOTE_CONFIGURATION_ENABLED
             value: {{ include "clusterAgent-remoteConfiguration-enabled" . | quote }}
-                    {{- if .Values.datadog.apm.instrumentation.enabled }}
+          {{- if .Values.datadog.apm.instrumentation.enabled }}
           - name: DD_APM_INSTRUMENTATION_ENABLED
             value: "true"
           {{- end }}
```

Comment:
As per [RFC](https://docs.google.com/document/d/12iU1nU50cGs27H5s2fUmUbxsjSpU2RO4HhAF1got2ds/edit?usp=sharing):
To allow more granular control of the injection process, Single Step Instrumentation introduces a set of new configuration parameters on the Cluster Agent:
`datadog.apm.instrumentation.enabled` - boolean indicating if automatic injection should be enabled in the whole cluster. Default value - false.
`datadog.apm.instrumentation.enabledNamespaces` - comma-separated list of namespaces where automatic injection is enabled. Default value - empty list. This setting applies only when cluster-level configuration datadog.apm.instrumentation.enabled is set to false. 
`datadog.apm.instrumentation.disabledNamespaces` - comma-separated list of namespaces where automatic injection needs to be disabled. Applies only when datadog.apm.instrumentation.enabled is set to true. Default value - empty list. Single Step Instrumentation will always skip injections in Kubernetes system namespace kube-system.

Reply:
This question was raised in Operator PR as well [here](https://github.com/DataDog/datadog-operator/pull/981#discussion_r1396156489).

Reply:
Hi @liliyadd 

I think the behavior should be changed, because all the other features in the agent don't use the same pattern, and IMO the current behavior is error-prone. 

What I think can make more sense is:
* `datadog.apm.instrumentation.enabled` - boolean indicating if automatic injection should be enabled
* `datadog.apm.instrumentation.enabledNamespaces` - comma-separated list of namespaces where automatic injection is enabled. (Default value - empty string). If value is empty, automatic injection is enabled on the whole cluster*. (*except `kube-systems`, and the "agent install" namespace).
* `datadog.apm.instrumentation.disabledNamespaces` - comma-separated list of namespaces where automatic injection needs to be disabled. (Default value - empty string). If `disabledNamespaces` automatic injection is active on every namespace except the one present in `disabledNamespaces` and `kube-systems`, and the "agent install" namespace.

I think this change can still be done, and make more sense for cross agent configuration genericity.